packages/react-slider/src/components/RangedSlider/RangedSlider.test.tsx

```diff
@@ -0,0 +1,602 @@
+    expect(onChange.mock.calls[0][1]).toEqual({ value: [17, 20] });
+    expect(lowerInputRef.current?.value).toBe('17');
+    expect(upperInputRef.current?.value).toBe('20');
+
+    fireEvent.keyDown(upperInputRef.current!, { key: 'ArrowUp' });
+    fireEvent.keyDown(upperInputRef.current!, { key: 'ArrowUp' });
+    expect(onChange.mock.calls[2][1]).toEqual({ value: [14, 17] });
+    expect(lowerInputRef.current?.value).toBe('14');
+    expect(upperInputRef.current?.value).toBe('17');
+
+    fireEvent.keyDown(upperInputRef.current!, { key: 'ArrowDown' });
+    fireEvent.keyDown(upperInputRef.current!, { key: 'ArrowDown' });
+    expect(onChange.mock.calls[4][1]).toEqual({ value: [14, 23] });
+    expect(lowerInputRef.current?.value).toBe('14');
+    expect(upperInputRef.current?.value).toBe('23');
+  });
+
+  it('handles a decimal step prop', () => {
+    const lowerInputRef = React.createRef<HTMLInputElement>();
+    const upperInputRef = React.createRef<HTMLInputElement>();
+    const onChange = jest.fn();
+
+    render(
+      <RangedSlider
+        defaultValue={[20, 20]}
+        step={0.0001}
+        onChange={onChange}
+        lowerInput={{ ref: lowerInputRef }}
+        upperInput={{ ref: upperInputRef }}
+      />,
+    );
+
+    fireEvent.keyDown(upperInputRef.current!, { key: 'ArrowUp' });
+    expect(onChange.mock.calls[0][1]).toEqual({ value: [20, 20.0001] });
+    expect(lowerInputRef.current?.value).toBe('20');
+    expect(upperInputRef.current?.value).toBe('20.0001');
+
+    fireEvent.keyDown(upperInputRef.current!, { key: 'ArrowDown' });
+    fireEvent.keyDown(upperInputRef.current!, { key: 'ArrowDown' });
+    expect(onChange.mock.calls[2][1]).toEqual({ value: [19.9999, 20] });
+    expect(lowerInputRef.current?.value).toBe('19.9999');
+    expect(upperInputRef.current?.value).toBe('20');
+  });
+
+  it('applies focus to lowerInput', () => {
+    const inputRef = React.createRef<HTMLInputElement>();
+    render(<RangedSlider lowerInput={{ ref: inputRef }} />);
+    inputRef?.current?.focus();
+    expect(document.activeElement).toEqual(inputRef.current);
+  });
+
+  it('applies focus to upperInput', () => {
+    const inputRef = React.createRef<HTMLInputElement>();
+    render(<RangedSlider upperInput={{ ref: inputRef }} />);
+    inputRef?.current?.focus();
+    expect(document.activeElement).toEqual(inputRef.current);
+  });
+
+  it('does not allow focus on disabled RangedSlider', () => {
+    const sliderRef = React.createRef<HTMLDivElement>();
+    const lowerInputRef = React.createRef<HTMLInputElement>();
+    const upperInputRef = React.createRef<HTMLInputElement>();
+
+    render(
+      <RangedSlider ref={sliderRef} disabled lowerInput={{ ref: lowerInputRef }} upperInput={{ ref: upperInputRef }} />,
+    );
+
+    expect(document.activeElement).toEqual(document.body);
+    sliderRef?.current?.focus();
+    lowerInputRef?.current?.focus();
+    upperInputRef?.current?.focus();
```

Comment:
We should have an `expect` call between each to ensure that focus didn't go back to the body just because it couldn't focus on the element we indicated.
```suggestion
    sliderRef?.current?.focus();
    expect(document.activeElement).toEqual(document.body);
    lowerInputRef?.current?.focus();
    expect(document.activeElement).toEqual(document.body);
    upperInputRef?.current?.focus();
```